axiom/nr_distributed_trace.c

```diff
@@ -481,14 +481,23 @@ void nr_distributed_trace_set_app_id(nr_distributed_trace_t* dt,
 }
 
 void nr_distributed_trace_set_trace_id(nr_distributed_trace_t* dt,
-                                       const char* trace_id) {
+                                       const char* trace_id,
+                                       bool do_padding) {
   if (NULL == dt) {
     return;
   }
 
   nr_free(dt->trace_id);
   if (trace_id) {
-    dt->trace_id = nr_strdup(trace_id);
+    int len = nr_strlen(trace_id);
```

Comment:
The call to `nr_strlen` is only needed when `do_padding` is `true`. There's no need to increase agent's overhead for default settings of the agent.

Reply:
changed in [e2c51c2](https://github.com/newrelic/newrelic-php-agent/pull/929/commits/e2c51c2595c8bd2620da4697790f957b9af4a4b4)